Source/WebCore/dom/Document.cpp

```diff
@@ -8095,10 +8095,16 @@ void Document::reveal()
     PageRevealEvent::Init init;
 
     // FIXME: This should implement more of https://drafts.csswg.org/css-view-transitions-2/#resolve-inbound-cross-document-view-transition
-    if (m_inboundViewTransitionParams && resolveViewTransitionRule())
-        init.viewTransition = ViewTransition::createInbound(*this, std::exchange(m_inboundViewTransitionParams, nullptr));
+    RefPtr<ViewTransition> inboundTransition;
+    if (m_inboundViewTransitionParams && resolveViewTransitionRule()) {
+        inboundTransition = ViewTransition::createInbound(*this, std::exchange(m_inboundViewTransitionParams, nullptr));
+        init.viewTransition = inboundTransition;
+    }
 
     dispatchWindowEvent(PageRevealEvent::create(eventNames().pagerevealEvent, WTFMove(init)), this);
+
+    if (inboundTransition)
+        inboundTransition->startInbound();
```

Comment:
I think we could remove the `startInbound` method and call `activateViewTransition` directly. It would also read more like the spec.